<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1243</width>
    <height>850</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_properties">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>260</y>
      <width>441</width>
      <height>511</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="title">
     <string>Properties</string>
    </property>
    <widget class="QScrollArea" name="scrollArea_properties">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>130</y>
       <width>431</width>
       <height>381</height>
      </rect>
     </property>
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents_properties">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>429</width>
        <height>379</height>
       </rect>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>411</width>
         <height>351</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_properties">
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>4</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
       </layout>
      </widget>
     </widget>
    </widget>
    <widget class="QPushButton" name="pushButton_properties">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>10</y>
       <width>75</width>
       <height>24</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="text">
      <string>确定</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_model">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>5</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
       <bold>false</bold>
       <underline>false</underline>
       <kerning>true</kerning>
      </font>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <item>
      <property name="text">
       <string>NO MODEL</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_position">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>Position</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_scale">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>Scale</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelPos_x">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelPos_y">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelPos_z">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelScale_x">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>80</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelScale_y">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>80</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelScale_z">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>80</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotation">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>Rotation</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelRotation_x">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>110</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelRotation_y">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>110</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_modelRotation_z">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>110</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_zTest">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>50</y>
       <width>79</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>zTest</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_zWrite">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>100</y>
       <width>79</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>zWrite</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_override">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>50</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="text">
      <string>overrider</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_cull">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>100</y>
       <width>68</width>
       <height>22</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>false</bold>
       <underline>false</underline>
      </font>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <item>
      <property name="text">
       <string>Back</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Front</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>None</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QLabel" name="label_canvas">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>800</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_light">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>20</y>
      <width>391</width>
      <height>241</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <italic>false</italic>
      <bold>true</bold>
      <underline>true</underline>
      <strikeout>false</strikeout>
     </font>
    </property>
    <property name="title">
     <string>Light</string>
    </property>
    <widget class="QLabel" name="label_light_rgb">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>44</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>RGB</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_r">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_g">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_b">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>50</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_light">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>200</y>
       <width>75</width>
       <height>24</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="text">
      <string>确定</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_light_intensity">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>Inten-</string>
     </property>
    </widget>
    <widget class="QSlider" name="slider_light_intensity">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>90</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>OpenHandCursor</cursorShape>
     </property>
     <property name="maximum">
      <number>30</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_light_intensity2">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>83</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_light_rgb_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>113</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>Dir</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_x">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>120</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_y">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>120</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_light_z">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>120</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_ambient_r">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>160</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_ambient_g">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>160</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_ambient_b">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>160</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>true</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
    </widget>
    <widget class="QLabel" name="label_light_ambient">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>155</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
       <italic>false</italic>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
      </font>
     </property>
     <property name="text">
      <string>Ambient</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1243</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_model">
    <property name="title">
     <string>Model</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_newmodel"/>
   </widget>
   <widget class="QMenu" name="menu_shader">
    <property name="title">
     <string>Shader</string>
    </property>
   </widget>
   <addaction name="menu_model"/>
   <addaction name="menu_shader"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string>New...</string>
   </property>
  </action>
  <action name="actionRemove">
   <property name="text">
    <string>Remove</string>
   </property>
  </action>
  <action name="action_newmodel">
   <property name="text">
    <string>New...</string>
   </property>
  </action>
  <action name="action_deleteModel">
   <property name="text">
    <string>Delete</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>lineEdit_light_r</tabstop>
  <tabstop>lineEdit_light_g</tabstop>
  <tabstop>lineEdit_light_b</tabstop>
  <tabstop>slider_light_intensity</tabstop>
  <tabstop>lineEdit_light_x</tabstop>
  <tabstop>lineEdit_light_y</tabstop>
  <tabstop>lineEdit_light_z</tabstop>
  <tabstop>lineEdit_ambient_r</tabstop>
  <tabstop>lineEdit_ambient_g</tabstop>
  <tabstop>lineEdit_ambient_b</tabstop>
  <tabstop>lineEdit_modelPos_x</tabstop>
  <tabstop>lineEdit_modelPos_y</tabstop>
  <tabstop>lineEdit_modelPos_z</tabstop>
  <tabstop>lineEdit_modelScale_x</tabstop>
  <tabstop>lineEdit_modelScale_y</tabstop>
  <tabstop>lineEdit_modelScale_z</tabstop>
  <tabstop>lineEdit_modelRotation_x</tabstop>
  <tabstop>lineEdit_modelRotation_y</tabstop>
  <tabstop>lineEdit_modelRotation_z</tabstop>
  <tabstop>checkBox_zTest</tabstop>
  <tabstop>checkBox_zWrite</tabstop>
  <tabstop>checkBox_override</tabstop>
  <tabstop>scrollArea_properties</tabstop>
  <tabstop>comboBox_cull</tabstop>
  <tabstop>comboBox_model</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
